FrontEnd/src/HomePage/MainPage.jsx

```diff
@@ -32,36 +67,41 @@ function MainPage({ userUID, userData, socket }) {
   return (
     <div className={`homePageList ${theme}`}>
       <div className="categorySection">
-        <CategoryList />
+        <CategoryList filterPosts={filterPosts} />
       </div>
-      <div className="feedSection">
-        {isLoading ? (
-          <p>Loading...</p>
-        ) : userFeed.length === 0 ? (
-          <p>No posts found.</p>
-        ) : (
-          userFeed.map((post, index) => (
-            <div
-              key={index}
-              className="postFeed"
-              onClick={() => toggleModal(post)}
-            >
-              <PostsPreview post={post} />
+      <div className="bodyArea">
+        <div className="searchArea">
+          <SearchBar filterPosts={filterPosts} />
+        </div>
+        <div className="feedSection">
+          {isLoading ? (
+            <p>Loading...</p>
+          ) : filteredPosts.length === 0 ? (
+            <p>No posts found.</p>
+          ) : (
+            filteredPosts.map((post, index) => (
+              <div
+                key={index}
```

Comment:
Use `post` ID (if available) instead of `index` for the `key` prop

Reply:
when I use that, it causes a console error and the list is buggy